app/App.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module App
+    ( Options(..)
+    , runProgram
+    ) where
+
+import qualified Data.Bifunctor as B
+import Control.Monad.Reader
+import Control.Monad.Except
+import Unused.Grouping (CurrentGrouping(..), groupedResponses)
+import Unused.Types (TermMatchSet, RemovalLikelihood(..))
+import Unused.TermSearch (SearchResults(..), fromResults)
+import Unused.ResponseFilter (withOneOccurrence, withLikelihoods, ignoringPaths)
+import Unused.Cache
+import Unused.TagsSource
+import Unused.ResultsClassifier
+import Unused.Aliases (termsAndAliases)
+import Unused.Parser (parseResults)
+import Unused.CLI (SearchRunner(..), renderHeader, executeSearch, withRuntime)
+import qualified Unused.CLI.Views as V
+
+type AppConfig = MonadReader Options
+
+data AppError
+    = TagError TagSearchOutcome
+    | InvalidConfigError [ParseConfigError]
+
+newtype App a = App {
+    runApp :: ReaderT Options (ExceptT AppError IO) a
+} deriving (Monad, Functor, Applicative, AppConfig, MonadError AppError, MonadIO)
```

Comment:
Yeah, I spotted that I was using `MonadReader Options m => m -> a` for pulling data out of `Options` (at the end of the file) and tried the type declaration. All it needed was `{-# LANGUAGE ConstraintKinds #-}`. Pretty neat.